tests/test_formatter.py

```diff
@@ -178,3 +181,48 @@ def test_internal_exception_formatting(caplog: pytest.LogCaptureFixture):
     assert len(caplog.records) == 1
     assert caplog.records[0].message.startswith('Caught an internal error in Logfire.')
     assert str(caplog.records[0].exc_info[1]) == 'bad scrubber'  # type: ignore
+
+
+@pytest.mark.skipif(sys.version_info[:2] == (3, 8), reason='fstring magic is only for 3.9+')
+@pytest.mark.anyio
+async def test_await_in_fstring(exporter: TestExporter):
+    """Test that logfire.info(f'{foo(await bar())}') evaluates the await expression and logs a warning."""
+
+    async def bar() -> str:
+        return 'content data'
+
+    def foo(x: str) -> str:
+        return x
+
+    with pytest.warns(FormattingFailedWarning) as warnings:
+        logfire.info(f'{foo(await bar())}')
+        [warning] = warnings
+        assert str(warning.message) == snapshot(
+            '\n'
+            '    Ensure you are either:\n'
+            '      (1) passing an f-string directly, with inspect_arguments enabled and working, or\n'
+            '      (2) passing a literal `str.format`-style template, not a preformatted string.\n'
```

Comment:
Following this advice won't help the user. KnownFormattingError says "In other words this should correspond to errors that would be raised when using `str.format`," which doesn't apply here. The general logic should be similar to KnownFormattingError but with a different exception class and warning message.